src/pipdeptree/_detect_env.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+import os
+import platform
+import subprocess  # noqa: S404
+import sys
+from pathlib import Path
+from typing import Callable
+
+
+def detect_active_interpreter() -> str:
+    """
+    Attempt to detect a venv, virtualenv, poetry, or conda environment by looking for certain markers.
+
+    If it fails to find any, it will fallback to using the system interpreter.
+    """
+    detection_funcs: tuple[Callable[[], Path | None]] = (
+        detect_venv_or_virtualenv_interpreter,
+        detect_conda_env_interpreter,
+        detect_poetry_env_interpreter,
+    )
+
+    for detect in detection_funcs:
+        path = detect()
+        if not path:
+            continue
+        if not path.exists():
+            break
+        return str(path)
+
+    return sys.executable
```

Comment:
I'm wondering if it makes sense to fallback to the system interpreter if we fail to detect any virtual environments?